Peer storage enhancements

adi2011 opened the pull request:
UX improvements and some other changes to make peer storage more efficient.

Addresses: #8382

adi2011 commented:
Hi @madelinevibes, sorry for the delay, I was AFK. I’ll try to complete this PR by the end of the week.

madelinevibes commented:
@adi2011 can you resolve the conflicts and clear CI by the end of this week so we can add this to 26.06? Release candidate planned for Monday 11 May. 

If you can't we'll move it to 25.09

adi2011 committed:
chanbackup: make getemergencyrecoverdata rpc more verbose

Adding more information to geremergencyrecoverdata, to let users know if they are using
legacy file format and the list of all the backed up channel ids.

Key Changes:
 - Added:
   1. can_create_penalty: To let user know if they need to update the file.
   2. backed_up_channel_ids: List of all the backed up channels

Changelog-Changed: Made getemergencyrecoverdata more verbose.

adi2011 committed:
chanbackup: Store the latest recvd peer storage only

Node should not store SCB that is older than what we already have.

Changelog-Fixed: Protocol: we now only store the most recent peer backup when recovering from peers.

rustyrussell commented:
Rebased, cleaned up a little.

adi2011 commented:
Thank you @rustyrussell! :orange_heart:
